backend/routes/domains.js

```diff
@@ -21,33 +21,61 @@ module.exports = function (router) {
 
       log.debug('Trying to verify DNS records of domain', domain, hostname)
 
+      // Verify the domain is RFC-valid
+      if (!domain.match(DNS_VALID)) {
+        log.debug(`${domain} is not a valid domain`)
+        return res.send({
+          success: true,
+          error: 'Invalid domain'
+        })
+      }
+
       try {
         const network = await Network.findOne({
           where: { networkId }
         })
 
         if (!network) {
-          return {
+          log.debug(`${networkId} is not a valid network for this node`)
+          return res.send({
             success: true,
             error: 'Invalid network ID'
-          }
+          })
         }
 
-        const networkConfig = getConfig(network.config)
+        const deployment = await ShopDeployment.findOne({
+          where: {
+            domain: {
+              [Op.like]: `%${domain}%`
+            }
+          },
+          order: [['created_at', 'DESC']],
+          limit: 1
```

Comment:
are order and limit necessary given findOne is called?

Reply:
Apparently limit is not.  Might be confusing this with another ORM(SqlAlchemy?) that will error if findOne returns more than one record.  Removed.  Order is necessary though since I want the newest only.